tests/functional/conftest.py

```diff
@@ -348,15 +353,16 @@ def _run_executable(self, prog, args, run_from_path, runtime):
         # Windows command prompt. Py.test is then able to collect stdout/sterr
         # messages and display them if a test fails.
 
-        process = psutil.Popen(args, executable=exe_path, stdout=sys.stdout,
-                               stderr=sys.stderr, env=prog_env, cwd=prog_cwd)
+        process = psutil.Popen(args, executable=exe_path, stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE, env=prog_env, cwd=prog_cwd)
```

Comment:
What is the idea behind this change?

Reply:
This is the other *one* of two problems. For some reason (as I explained in the previous PR that attempted this), the subprocess module will "irritate" the stdout handle by making various C API calls to the underlying handle (which is nonstandard/provided by pytest), such that it closes non-deterministically. Once that happens, the subprocess will die when it can no longer write to stdout or stderr.